src/main/java/com/conveyal/datatools/editor/controllers/api/CalendarController.java

```diff
@@ -62,54 +58,44 @@ public static Object getCalendar(Request req, Response res) {
             }
             else if (patternId != null) {
                 if (!tx.tripPatterns.containsKey(patternId)) {
-                    tx.rollback();
                     halt(404);
                 }
 
                 Set<String> serviceCalendarIds = Sets.newHashSet();
-                for (Trip trip : tx.getTripsByPattern(patternId)) {
-                    serviceCalendarIds.add(trip.calendarId);
-                }
+                serviceCalendarIds.addAll(tx.getTripsByPattern(patternId).stream()
+                        .map(trip -> trip.calendarId)
+                        .collect(Collectors.toList()));
 
-                Collection<ServiceCalendarForPattern> ret =
-                        Collections2.transform(serviceCalendarIds, new Function<String, ServiceCalendarForPattern>() {
-
-                            @Override
-                            public ServiceCalendarForPattern apply(String input) {
-                                ServiceCalendar cal = tx.calendars.get(input);
-
-                                Long count = tx.tripCountByPatternAndCalendar.get(new Fun.Tuple2(patternId, cal.id));
-
-                                if (count == null) count = 0L;
-
-                                return new ServiceCalendarForPattern(cal, tx.tripPatterns.get(patternId), count);
-                            }
-
-                        });
+                Collection<ServiceCalendarForPattern> ret = new HashSet<>();
 
+                for (String calendarId : serviceCalendarIds) {
+                    ServiceCalendar cal = tx.calendars.get(calendarId);
+                    Long count = tx.tripCountByPatternAndCalendar.get(new Fun.Tuple2(patternId, cal.id));
+                    if (count == null) count = 0L;
+                    ret.add(new ServiceCalendarForPattern(cal, tx.tripPatterns.get(patternId), count));
+                }
                 return ret;
             }
             else {
-                Collection<ServiceCalendar> cals = tx.calendars.values();
+                Collection<ServiceCalendar> cals = new ArrayList<>(tx.calendars.values());
```

Comment:
Not to be nitpicky, but comments starting with `/**` are Javadoc comments (although since it's not above a class/function/field def or at the top of the file it shouldn't be much of a problem, whereas `//` and `/*` are standard comments. I generally only use `//` comments in code, because then you can comment out a whole block of code with the `/*` style without wondering if there are other `/*` comments in there to throw a wrench in things.